vnet/lib/vnet/node_api/network.rb

```diff
@@ -40,9 +40,9 @@ def dispatch_deleted_item_events(model)
 
         # 0001_origin
         ActiveNetwork.dispatch_deleted_where(filter, model.deleted_at)
-        # IpAddresses.dispatch_deleted_where(filter, model.deleted_at) # Needed? We're deleting the network.
         DatapathNetwork.dispatch_deleted_where(filter, model.deleted_at)
         Route.dispatch_deleted_where(filter, model.deleted_at)
+        model.ip_leases.each {  |lease| IpLease.dispatch_deleted_for_model(lease) }
```

Comment:
This should probably be moved to IpLeases, perhaps a special dispatch method.

Note that when you call model.ip_leases here, the ip_leases already have is_deleted set, so it won't actually find any ip leases. You need to use the same kind of sequel tricks that dispatch_deleted_where does to get the list of ip_leases.